src/common/mfu_daos.c

```diff
@@ -1530,7 +1529,7 @@ int daos_cleanup(
     MPI_Barrier(MPI_COMM_WORLD);
 
     /* Unmount destination DFS container */
-    if ((mfu_dst_file != NULL) && (mfu_dst_file->dfs != NULL)) {
```

Comment:
@daltonbohning is the first NULL check no longer necessary?

Reply:
@dsikich Not sure why I did that. I think it's safer to keep the first NULL check. I'll fix that, and also add it for the src_file above

Reply:
@dsikich I moved the NULL checks to the helper function and also reorganized a bit so we could remove some barriers.